slc/component/ot_crash_handler.slcc

```diff
@@ -0,0 +1,65 @@
+id: ot_crash_handler
+label: Crash Handler
+package: OpenThread
+category: OpenThread
+quality: production
+description: |-
+  This component provides a set of APIs for printing crash info.
+  In the case of a crash, this component captures the details.
+  The provided `efr32PrintResetInfo()` API prints the crash details.
+provides:
+  - name: ot_crash_handler
+conflicts:
+  - name: legacy_hal_soc
+requires:
+  - name: component_catalog
+  - name: device
+  - name: emlib_rmu
+  - name: event_handler
+  # The vendor extension component will print the crash info after the OpenThread instance is initialized.
+  - name: ot_vendor_extension
+include:
+  - path: third_party/silabs/gecko_sdk/protocol/openthread/src/legacy_hal/include
+    file_list:
+      - path: crash_handler.h
+  - path: third_party/silabs/gecko_sdk/platform/service/legacy_hal/inc
+    file_list:
+    - path: asm.h
+    - path: efm32_micro.h
+      condition:
+        - device_cortexm
+    - path: micro-common.h
+    - path: micro-types.h
+    - path: micro.h
+    - path: reset-def.h
+    - path: platform-header.h
+    - path: cortexm3/diagnostic.h
+      condition:
+      - device_cortexm
+source:
+  - path: third_party/silabs/gecko_sdk/platform/service/legacy_hal/src/faults.s
+  - path: third_party/silabs/gecko_sdk/protocol/openthread/src/legacy_hal/crash_handler.c
+  - path: src/legacy_hal/diagnostic.c
+define:
+  - name: "PLATFORM_HEADER"
+    value: "\"platform-header.h\""
```

Comment:
What is this header file, and why is it being defined here?

Reply:
The header file refers to [platform/service/legacy_hal/inc/platform-header.h](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/platform/service/legacy_hal/inc/platform-header.h)

This component was originally created based off [platform/service/component/legacy_hal_soc.slcc](https://github.com/SiliconLabs/gecko_sdk/blob/gsdk_4.4/platform/service/component/legacy_hal_soc.slcc) with everything stripped out except for the bare-minimum required for the crash handler